### PR TITLE
fix: add FlattenVector transformer

### PR DESCRIPTION
**Pull request by s0nicboOm**
1. Fix saving of stateless transformer
2. Introduce FlattenVector

### CODE REVIEW

File: numalogic/transforms/_stateless.py [hunk 1]
```diff
@@ -53,3 +53,22 @@ def transform(self, X, **__):
...
+class FlattenVector(StatelessTransformer):
+    """A stateless transformer that flattens a vector.
+
+    Args:
+    ____
+        n_features: number of features
+
+    """
+
+    def __init__(self, n_features):
```
Comment (ab93) [inline numalogic/transforms/_stateless.py:67]:
```suggestion
    def __init__(self, n_features: int):
```

File: numalogic/udfs/_config.py [hunk 2]
```diff
@@ -44,6 +44,7 @@ class MLPipelineConf:
 
     pipeline_id: str = "default"
     metrics: list[str] = field(default_factory=list)
+    combine_metrics: bool = False
```
Comment (ab93) [inline numalogic/udfs/_config.py:47]: what is this for?

File: tests/registry/test_redis_registry.py [hunk 3]
```diff
@@ -196,8 +196,8 @@ def test_multiple_save(self):
         self.registry.save_multiple(
             skeys=self.skeys,
             dict_artifacts={
-                "AE": KeyedArtifact(dkeys=["AE"], artifact=VanillaAE(10)),
-                "scaler": KeyedArtifact(dkeys=["scaler"], artifact=StandardScaler()),
+                "AE": KeyedArtifact(dkeys=["AE"], artifact=VanillaAE(10), stateful=True),
```
Comment (ab93) [inline tests/registry/test_redis_registry.py:199]: do we need to provide this everytime?

File: numalogic/transforms/_stateless.py (hunk 1 above)
Comment (s0nicboOm) [reply numalogic/transforms/_stateless.py:67]: let me add this

File: numalogic/udfs/_config.py (hunk 2 above)
Comment (s0nicboOm) [reply numalogic/udfs/_config.py:47]: for instance: [failed, degraded]

now all the metrics, pipeline payload are looking for 1 feature after the transform. Hence, this was needed.

File: tests/registry/test_redis_registry.py (hunk 3 above)
Comment (s0nicboOm) [reply tests/registry/test_redis_registry.py:199]: What do you think about it? The other option is set a default value for it

File: numalogic/udfs/_config.py (hunk 2 above)
Comment (s0nicboOm) [reply numalogic/udfs/_config.py:47]: This get's checked in preprocess step and based on the value we combine the metrics name too

File: tests/registry/test_redis_registry.py (hunk 3 above)
Comment (s0nicboOm) [reply tests/registry/test_redis_registry.py:199]: This should inherently be assigned.

Comment (s0nicboOm) [reply tests/registry/test_redis_registry.py:199]: For redis registry it is not used anywhere, so I am thinking to make it optional value.

File: numalogic/udfs/preprocess.py [hunk 4]
```diff
@@ -174,6 +174,15 @@ def exec(self, keys: list[str], datum: Datum) -> Messages:
...
             x_scaled = self.compute(model=preproc_clf, input_=payload.get_data())
+
+            # make one feature vector
+            if x_scaled.shape[1] != len(payload.metrics) and x_scaled.shape[1] == 1:
+                payload = replace(payload, metrics=["-".join(payload.metrics)])
+                _LOGGER.info(
+                    "%s - Metrics used: %s",
+                    payload.uuid,
+                    payload.metrics,
+                )
```
Comment (ab93) [inline numalogic/udfs/preprocess.py:185]: Can you do it in a separate function?

File: pyproject.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "numalogic"
-version = "0.6.1"
+version = "0.6.2a1"
```
Comment (ab93) [inline pyproject.toml:3]: if you have tested it already make it 0.6.2, and cut a release

**Review by ab93 — COMMENTED** [review]
Minor comment

File: pyproject.toml (hunk 5 above)
Comment (s0nicboOm) [reply pyproject.toml:3]: got it

**Review by ab93 — APPROVED** [review]

**Pull request merged**